server/sonar-server/src/main/java/org/sonar/server/computation/measure/MeasureRepositoryImpl.java

```diff
@@ -173,4 +237,55 @@ public String apply(@Nonnull BatchReport.Measure input) {
       return input.getMetricKey();
     }
   }
+
+  @Immutable
+  private static final class MeasureKey {
+    private static final int DEFAULT_INT_VALUE = -6253;
```

Comment:
What is that ? 
Why not use a Object for ruleId and characteristicId, and set default value to null ?